rfcs/0122-system-permission.md

```diff
@@ -0,0 +1,712 @@
+# Motivation
+
+The `Administrator` role currently combines both project administration functionality and system administration functionality. Giving a user access to do things like delete a signal or create a shared API key cannot be done without also giving them access to do things like installing apps and modifying the Seq license. 
+
+The Seq permissions system currently has four permissions, Read, Write, Setup, and Ingest. These are allocated to four built in roles, Administrator, User (read-only), User (read/write) and User (read/write/ingest).
+
+The Administrator role currently encompasses both project administration, and host system access, through functionality such as Seq App installation.
+
+This RFC proposes splitting the Setup permission into Project (project administration) and System (host system administration), so that these two responsibilities can be separated. Team leaders often need access to project administration features such as API keys and retention policies, but don't necessarily need privileged access to the host system.
+
+The built in Administrator role will include both Project and System, i.e., retaining the functionality of this role today. A new role Project Owner will have Project permissions, but not System.
+
+# Detailed Design
+
+## Roles to Permission Mapping
+
+| Role | 2021.4 permissions | 2022.1 permissions |
+| -- | -- | -- |
+| User (read-only) | Read | Read |
+| User (read/write) | Read, Write | Read, Write |
+| User (read/write/ingest) | Read, Write, Ingest | Read, Write, Ingest |
+| Project Owner | | Read, Write, Ingest, Project |
+| Administrator | Read, Write, Ingest, Setup | Read, Write, Ingest, Project, System |
+
+## 2021.4 Permission to API Endpoint Mapping
```

Comment:
A brief summary of the kinds of things accessible with each permission might help, here? E.g. System permission will include configuring feeds, apps, diagnostics, telemetry/update checking opt-in, ...; Project permission will include retention, API keys, app instance configuration, suppression of notifications, ....

Reply:
should retention policies be `System`?

Reply:
I don't think so. They don't grant any access to or reveal any information about the host system that Seq is running on.